main.tsp

```diff
@@ -1,278 +0,0 @@
-import "@typespec/http";
-import "@typespec/rest";
-import "@typespec/openapi3";
-
-using Http;
-
-@service({
-  title: "Todo App",
-})
-@useAuth(BearerAuth)
-namespace Todo;
-
-model User {
-  /** An autogenerated unique id for the user */
-  @key
-  @visibility("read")
-  id: safeint;
-
-  /** The user's username */
-  @maxLength(50)
-  username: string;
-
-  /** The user's email address */
-  @format("email")
-  email: string;
-
-  /**
-   * The user's password, provided when creating a user
-   * but is otherwise not visible (and hashed by the backend)
-   */
-  @visibility("create")
-  password: string;
-}
-
-model TodoItem {
-  /** The item's unique id */
-  @key id: safeint;
-
-  /** The item's title */
-  @maxLength(255)
-  title: string;
-
-  /** User that created the todo */
-  @visibility("read") createdBy: User.id;
-
-  /** User that the todo is assigned to */
-  ownedBy: User.id;
-
-  /** A longer description of the todo item in markdown format */
-  description: string;
```

Comment:
This probably should be optional? If nothing else because there are relatively few optional model fields in the spec, and having a couple would be goodness...